### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by allenwq**
Remove course discussion topic factory
Cleanup code according to the latest rubocop

### CODE REVIEW

File: spec/models/course/discussion/post_spec.rb [hunk 1]
```diff
@@ -41,9 +41,8 @@
         expect(root_post.first).to eq(graph[:root])
 
         root_children = root_post.second
-        expect(root_children).to contain_exactly([graph[:a], [
-                                                   [graph[:b], []]]],
-                                                 [graph[:c], []])
+        expect(root_children).
+          to contain_exactly([graph[:a], [[graph[:b], []]]], [graph[:c], []])
```
Comment (weiqingtoh) [inline spec/models/course/discussion/post_spec.rb]: This is a little hard to parse - not sure if we want to represent this visually using either some indentation here, or we can do it using a variable or comments, or using `Array.new(...)`. Let me see if I can come up with something more readable.

Comment (weiqingtoh) [inline spec/models/course/discussion/post_spec.rb]: This is the best I can come up with >_< 

If this is not an improvement in readability we can just ignore this. 

```ruby
let(:root_children_graph) do 
  [graph[:a], [[graph[:b], []]]], [graph[:c], []]
  # [ graph[:a],
  #   [ 
  #     [ graph[:b], [ ] 
  #     ] 
  #   ]
  # ], [ graph[:c], [ ] ]
end
...
  expect(root_children).to contain_exactly(root_children_graph)
```

**Comment by weiqingtoh** [comment]
For the fixes to Rubocop - [Multiline Method Call Brace Layout](http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/MultilineMethodCallBraceLayout): I assume we are moving to `symmetrical` or `new_line`?

If yes, should we add [this](https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L679-L687) to the `.rubocop.yml` file?

File: spec/models/course/discussion/post_spec.rb (hunk 1 above)
Comment (allenwq) [inline spec/models/course/discussion/post_spec.rb]: why not just use the style in your comments  ? that is:
```
let(:root_children_graph) do 
   [ graph[:a],
     [ 
       [
         graph[:b], [] 
       ] 
     ]
   ], [graph[:c], []]
end
```

**Comment by allenwq** [comment]
@weiqingtoh `symmetrical` is the default, I think we don't need to change anything ?

File: spec/models/course/discussion/post_spec.rb (hunk 1 above)
Comment (weiqingtoh) [inline spec/models/course/discussion/post_spec.rb]: 👍

**Comment by lowjoel** [comment]
I don't see the factory removed? I'm not entirely for removing it, for reasons outlined in the other PR.

**Comment by allenwq** [comment]
@lowjoel  Finally I didn't remove the factory after discussion. yah there are some good parts of having it, especially we can put common traits for all the discussion topics there.

**Comment by lowjoel** [comment]
:+1:
